src/centered-on-asteroid/train.py

```diff
@@ -141,27 +148,227 @@ def train(
+        type=int,
+        default=10,
+        help="The output (feature vector) size of the CNN (default 10).",
+    )
+
+    parser.add_argument(
+        "--num_conv_blocks",
+        type=int,
+        default=2,
+        help="The number of convolutional blocks in the CNN (default 2).",
+    )
+
+    parser.add_argument(
+        "--conv_filters_list",
+        type=str,
+        default="16,32",
+        help="The number of filters in each convolutional block (default '16,32'). There should be num_conv_blocks separated by commas.",
+    )
+
+    parser.add_argument(
+        "--conv_kernel_sizes",
+        type=int,
+        default=3,
+        help="The kernel size of the convolutional layers (default 3).",
+    )
+
+    parser.add_argument(
+        "--conv_strides",
+        type=int,
+        default=1,
+        help="The stride of the convolutional layers (default 1).",
+    )
+
+    parser.add_argument(
+        "--conv_padding",
+        type=int,
+        default=1,
+        help="The padding of the convolutional layers (default 1).",
+    )
+
+    # MLP parameters
+
+    parser.add_argument(
+        "--num_hidden_mlp_layers",
+        type=int,
+        default=2,
+        help="The number of hidden layers in MLP (default 2).",
+    )
+
+    parser.add_argument(
+        "--hidden_mlp_size",
+        type=int,
+        default=64,
+        help="The size of the hidden layers in MLP (default 64).",
+    )
+    # Hyperparameters
+    parser.add_argument(
+        "--loss",
+        type=str,
+        default="BCELoss",
+        help="The loss function to use (Default BCELoss). Choices are: CrossEntropyLoss, NLLLoss, BCELoss, BCEWithLogitsLoss, MultiLabelSoftMarginLoss, MultiMarginLoss, SoftMarginLoss.",
+    )
+    parser.add_argument(
+        "--optimizer",
+        type=str,
+        default="Adam",
+        help="The optimizer to use (Default Adam).",
+    )
+    parser.add_argument(
+        "--epochs",
+        type=int,
+        default=10,
+        help="Total number of epochs to train for (default 10).",
+    )
+    parser.add_argument(
+        "--batch_size", type=int, default=4, help="Batch size (default 4)."
+    )
+    parser.add_argument(
+        "--experiment_name",
+        type=str,
+        default="end-to-end",
+        help="Name of the experiment (This will be combined with a timestamp to create a unique identifier). Default is 'end-to-end'.",
+    )
 
-    model = model_def.CFN(images_per_sequence, feature_vector_size, image_shape)
+    # Data
+    parser.add_argument(
+        "--metadata",
+        type=str,
+        default="no_metadata",
+        help="Metadata to use. Default is 'no_metadata'. Choices are: 'no_metadata', 'max_grad_diff', 'max_ang_diff', 'max_movement_vector_distance', 'max_movement_vector_distance_normalised', 'gradients', 'angles', 'movement_vectors'.",
+    )
+    parser.add_argument(
+        "--data_path",
+        type=str,
+        default="./../../scripts/data",
```

Comment:
wrong folder. Also maybe extract all of the default paths to some config file so its reusable